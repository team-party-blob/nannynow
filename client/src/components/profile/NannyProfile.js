import React, { PureComponent, Fragment } from 'react';
import PropTypes from 'prop-types';
import styles from './Profile.css';
import nannyProfileStyles from './NannyProfile';
import globalStyles from '../../main.css';
import { ROUTES } from '../../routes/routes';
import { statesArray } from './helpers/statesArray';

export default class NannyProfile extends PureComponent {
  static propTypes = {
    session: PropTypes.object.isRequired,
    profile: PropTypes.object,
    createProfile: PropTypes.func.isRequired,
    updateProfile: PropTypes.func.isRequired
  };

  state = {
    photo: '',
    name: '',
    age: 0,
    streetAddress1: '',
    streetAddress2: '',
    city: '',
    state: '',
    zip: '',
    phone: '',
    pricePerHour: 16,
    description: ''
  };

  componentDidMount() {
    const { profile } = this.props;
    this.setState(profile);
  }

  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  };

  handleSubmit = event => {
    event.preventDefault();
    const { session, profile, updateProfile, createProfile } = this.props;
    const { _id, agency } = session;
    const profileInfo = { ...this.state, user: _id, agency };
    {
      profile && updateProfile(profile._id, profileInfo);
    }
    {
      !profile && createProfile(profileInfo);
    }
    // this.props.history.push(ROUTES.DASHBOARD.linkTo(_id));
  };

  render() {
    const {
      photo,
      name,
      age,
      streetAddress1,
      streetAddress2,
      state,
      city,
      zip,
      phone,
      pricePerHour,
      description
    } = this.state;

    const statesList = statesArray.map(stateItem => {
      return (
        <option key={stateItem} value={stateItem}>
          {stateItem}
        </option>
      );
    });

    return (
      <Fragment>
        <h1 id={globalStyles.header}>Create Your Profile</h1>
        <div id={styles.profileBody}>
          <form onSubmit={this.handleSubmit} id={styles.profileForm}>
            <label htmlFor='name'>Full Name:</label>
            <input
              type='text'
              name='name'
              value={name}
              onChange={this.handleChange}
              required
            />
            <label htmlFor='photo'>Upload A Photo:</label>
            <input
              type='url'
              name='photo'
              value={photo}
              onChange={this.handleChange}
            />
            <label htmlFor='age'>Age:</label>
            <input
              type='number'
              name='age'
              value={age}
              onChange={this.handleChange}
            />
            <label htmlFor='streetAddress1'>Street Address:</label>
            <input
              type='text'
              name='streetAddress1'
              value={streetAddress1}
              onChange={this.handleChange}
              required
            />
            <input
              type='text'
              name='streetAddress2'
              value={streetAddress2}
              onChange={this.handleChange}
            />
            <label htmlFor='city'>City</label>
            <input
              type='text'
              name='city'
              value={city}
              onChange={this.handleChange}
              required
            />
            <label htmlFor='state'>State</label>
            <select
              name='state'
              value={state}
              onChange={this.handleChange}
              required
            >
              <option value=''>--</option>
              {statesList}
            </select>
            <label htmlFor='zip'>Zip</label>
            <input
              type='text'
              name='zip'
              value={zip}
              onChange={this.handleChange}
              required
            />
            <label htmlFor='phone'>Phone Number</label>
            <input
              type='text'
              name='phone'
              value={phone}
              onChange={this.handleChange}
              required
            />
            <label htmlFor='pricePerHour'>Your Hourly Rate:</label>
            <input
              type='number'
              name='pricePerHour'
              value={pricePerHour}
              onChange={this.handleChange}
            />
            <label htmlFor='description'>Tell Us About Yourself:</label>
            <textarea
              type='text'
              name='description'
              value={description}
              onChange={this.handleChange}
            />
            <button id={styles.submitButton}>Submit Profile</button>
          </form>
          <div id={styles.profileView}>
            <div id={styles.doubleBorder1}>
              <div id={styles.doubleBorder2}>
                <div id={styles.profileHeader}>
                  <img src={photo} />
                  <h1>{name}</h1>
                </div>
                <h3>Age: {age}</h3>
                <h3>Street Address: {streetAddress1}</h3>
                <h3>{streetAddress2}</h3>
                <h3>City: {city}</h3>
                <h3>State: {state}</h3>
                <h3>Zip Code: {zip}</h3>
                <h3>Phone Number: {phone}</h3>
                <h3>Hourly Rate: {pricePerHour}</h3>
                <h3>Description: {description}</h3>
              </div>
            </div>
          </div>
        </div>
      </Fragment>
    );
  }
}
