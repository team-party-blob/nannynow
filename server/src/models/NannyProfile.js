import mongoose from 'mongoose';
import { getStates } from './helpers/listHelpers';

const nannyProfileSchema = new mongoose.Schema({
  user: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  agency: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Agency',
    required: true
  },
  name: {
    type: String,
    required: true
  },
  photo: {
    type: String
  },
  streetAddress1: {
    type: String,
    required: [true, 'Street address is required.']
  },
  streetAddress2: String,
  city: {
    type: String,
    required: [true, 'City is required.']
  },
  state: {
    type: String,
    required: [true, 'State is required.'],
    enum: getStates()
  },
  zip: {
    type: String,
    required: [true, 'ZIP code is required.']
  },
  phone: {
    type: String,
    required: [true, 'State is required.']
  },
  description: {
    type: String
  },
  age: {
    type: Number
  },
  pricePerHour: {
    type: Number
  },
  createdDate: {
    type: Date,
    default: Date.now()
  }
});

export default mongoose.model('NannyProfile', nannyProfileSchema);
