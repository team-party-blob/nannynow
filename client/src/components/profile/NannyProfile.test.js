import React from 'react';
import { shallow } from 'enzyme';
import NannyProfile from './NannyProfile';

describe('Profile component', () => {
  it('renders a component', () => {
    const wrapper = shallow(<NannyProfile />);
    expect(wrapper).toMatchSnapshot();
  });



});
