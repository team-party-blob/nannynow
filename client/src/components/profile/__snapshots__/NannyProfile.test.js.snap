// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component renders a component 1`] = `
<Fragment>
  <h1
    id="profileHeader"
  >
    Create Your Profile
  </h1>
  <div
    id="profileBody"
  >
    <form
      onSubmit={[Function]}
    >
      <label
        htmlFor="name"
      >
        Full Name:
      </label>
      <input
        name="name"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
      <label
        htmlFor="photo"
      >
        Upload A Photo:
      </label>
      <input
        name="photo"
        onChange={[Function]}
        type="file"
        value=""
      />
      <label
        htmlFor="age"
      >
        Age:
      </label>
      <input
        name="age"
        onChange={[Function]}
        type="number"
        value={0}
      />
      <label
        htmlFor="streetAddress1"
      >
        Street Address:
      </label>
      <input
        name="streetAddress1"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
      <input
        name="streetAddress2"
        onChange={[Function]}
        type="text"
        value=""
      />
      <label
        htmlFor="city"
      >
        City
      </label>
      <input
        name="city"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
      <label
        htmlFor="state"
      >
        State
      </label>
      <select
        name="state"
        onChange={[Function]}
        required={true}
        value=""
      >
        <option
          value=""
        >
          --
        </option>
        <option
          key="AL"
          value="AL"
        >
          AL
        </option>
        <option
          key="AK"
          value="AK"
        >
          AK
        </option>
        <option
          key="AS"
          value="AS"
        >
          AS
        </option>
        <option
          key="AZ"
          value="AZ"
        >
          AZ
        </option>
        <option
          key="AR"
          value="AR"
        >
          AR
        </option>
        <option
          key="CA"
          value="CA"
        >
          CA
        </option>
        <option
          key="CO"
          value="CO"
        >
          CO
        </option>
        <option
          key="CT"
          value="CT"
        >
          CT
        </option>
        <option
          key="DE"
          value="DE"
        >
          DE
        </option>
        <option
          key="DC"
          value="DC"
        >
          DC
        </option>
        <option
          key="FM"
          value="FM"
        >
          FM
        </option>
        <option
          key="FL"
          value="FL"
        >
          FL
        </option>
        <option
          key="GA"
          value="GA"
        >
          GA
        </option>
        <option
          key="GU"
          value="GU"
        >
          GU
        </option>
        <option
          key="HI"
          value="HI"
        >
          HI
        </option>
        <option
          key="ID"
          value="ID"
        >
          ID
        </option>
        <option
          key="IL"
          value="IL"
        >
          IL
        </option>
        <option
          key="IN"
          value="IN"
        >
          IN
        </option>
        <option
          key="IA"
          value="IA"
        >
          IA
        </option>
        <option
          key="KS"
          value="KS"
        >
          KS
        </option>
        <option
          key="KY"
          value="KY"
        >
          KY
        </option>
        <option
          key="LA"
          value="LA"
        >
          LA
        </option>
        <option
          key="ME"
          value="ME"
        >
          ME
        </option>
        <option
          key="MH"
          value="MH"
        >
          MH
        </option>
        <option
          key="MD"
          value="MD"
        >
          MD
        </option>
        <option
          key="MA"
          value="MA"
        >
          MA
        </option>
        <option
          key="MI"
          value="MI"
        >
          MI
        </option>
        <option
          key="MN"
          value="MN"
        >
          MN
        </option>
        <option
          key="MS"
          value="MS"
        >
          MS
        </option>
        <option
          key="MO"
          value="MO"
        >
          MO
        </option>
        <option
          key="MT"
          value="MT"
        >
          MT
        </option>
        <option
          key="NE"
          value="NE"
        >
          NE
        </option>
        <option
          key="NV"
          value="NV"
        >
          NV
        </option>
        <option
          key="NH"
          value="NH"
        >
          NH
        </option>
        <option
          key="NJ"
          value="NJ"
        >
          NJ
        </option>
        <option
          key="NM"
          value="NM"
        >
          NM
        </option>
        <option
          key="NY"
          value="NY"
        >
          NY
        </option>
        <option
          key="NC"
          value="NC"
        >
          NC
        </option>
        <option
          key="ND"
          value="ND"
        >
          ND
        </option>
        <option
          key="MP"
          value="MP"
        >
          MP
        </option>
        <option
          key="OH"
          value="OH"
        >
          OH
        </option>
        <option
          key="OK"
          value="OK"
        >
          OK
        </option>
        <option
          key="OR"
          value="OR"
        >
          OR
        </option>
        <option
          key="PW"
          value="PW"
        >
          PW
        </option>
        <option
          key="PA"
          value="PA"
        >
          PA
        </option>
        <option
          key="PR"
          value="PR"
        >
          PR
        </option>
        <option
          key="RI"
          value="RI"
        >
          RI
        </option>
        <option
          key="SC"
          value="SC"
        >
          SC
        </option>
        <option
          key="SD"
          value="SD"
        >
          SD
        </option>
        <option
          key="TN"
          value="TN"
        >
          TN
        </option>
        <option
          key="TX"
          value="TX"
        >
          TX
        </option>
        <option
          key="UT"
          value="UT"
        >
          UT
        </option>
        <option
          key="VT"
          value="VT"
        >
          VT
        </option>
        <option
          key="VI"
          value="VI"
        >
          VI
        </option>
        <option
          key="VA"
          value="VA"
        >
          VA
        </option>
        <option
          key="WA"
          value="WA"
        >
          WA
        </option>
        <option
          key="WV"
          value="WV"
        >
          WV
        </option>
        <option
          key="WI"
          value="WI"
        >
          WI
        </option>
        <option
          key="WY"
          value="WY"
        >
          WY
        </option>
      </select>
      <label
        htmlFor="zip"
      >
        Zip
      </label>
      <input
        name="zip"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
      <label
        htmlFor="phone"
      >
        Phone Number
      </label>
      <input
        name="phone"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
      <label
        htmlFor="pricePerHour"
      >
        Your Hourly Rate:
      </label>
      <input
        name="pricePerHour"
        onChange={[Function]}
        type="number"
        value={16}
      />
      <label
        htmlFor="description"
      >
        Tell Us About Yourself:
      </label>
      <input
        name="description"
        onChange={[Function]}
        type="text"
        value=""
      />
      <button>
        Submit Profile
      </button>
    </form>
    <div
      id="profileView"
    >
      <img
        src=""
      />
      <h1>
        Name: 
      </h1>
      <h3>
        Age: 
        0
      </h3>
      <h3>
        Street Address:
      </h3>
      <h3>
        Address (continued): 
      </h3>
      <h3>
        City: 
      </h3>
      <h3>
        State: 
      </h3>
      <h3>
        Zip Code: 
      </h3>
      <h3>
        Phone Number: 
      </h3>
      <h3>
        Hourly Rate: 
        16
      </h3>
      <h3>
        Description: 
      </h3>
    </div>
  </div>
</Fragment>
`;
